### PR TITLE
Improve test coverage for `model` package

### PR DESCRIPTION
**Pull request by chaaaaun**

### CODE REVIEW

File: src/main/java/scrolls/elder/model/LogStore.java [hunk 1]
```diff
@@ -107,10 +107,12 @@ public void addLog(Log newLog) {
      * Adds an existing log to the store.
      * For the case where existing logs are read from storage.
      */
-    public void addLogWithId(Log l) {
-        logs.put(l.getLogId(), l);
-        if (l.getLogId() > logIdSequence) {
-            logIdSequence = l.getLogId() + 1;
+    public void addLogWithId(Log newLog) {
+        requireNonNull(newLog);
```
Comment (gerteck) [inline src/main/java/scrolls/elder/model/LogStore.java:111]: I like the use of `requireNonNull` that throws `NullPointerException` for doing parameter validation, helps increase robustness of the system 😄

File: src/test/java/scrolls/elder/model/log/LogTest.java [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+        int hashCode2 = log1.hashCode();
+        int hashCode3 = log2.hashCode();
+
+        // Consistency: hashCode on same object returns same result
+        assertEquals(hashCode1, hashCode2);
+
+        // Equality: equal objects return same hashCode
+        assertEquals(hashCode1, hashCode3);
+    }
+}
```
Comment (gerteck) [inline src/test/java/scrolls/elder/model/log/LogTest.java:60]: Would be good to add a test toStringMethod, such as the following:
    
```
    @Test
    public void testToString() {
        String expectedToStringAliceToElle = "Log ID: 0\n" +
                                             "Volunteer ID: 0\n" +
                                             "Befriendee ID: 4\n" +
                                             "Duration: 2\n" +
                                             "Start Date: Mon Mar 07 00:00:00 UTC 2024\n" +
                                             "Remarks: Was great!";

        String expectedToStringBensonToFiona = "Log ID: 1\n" +
                                               "Volunteer ID: 1\n" +
                                               "Befriendee ID: 5\n" +
                                               "Duration: 3\n" +
                                               "Start Date: Fri Apr 08 00:00:00 UTC 2024\n" +
                                               "Remarks: Was okay.";

        assertEquals(expectedToStringAliceToElle, TypicalLogs.LOG_ALICE_TO_ELLE.toString());
        assertEquals(expectedToStringBensonToFiona, TypicalLogs.LOG_BENSON_TO_FIONA.toString());
    }
```

Everything looks good! 👍

**Review by gerteck — APPROVED** [review]
Looks great to me!

File: src/test/java/scrolls/elder/model/log/LogTest.java (hunk 2 above)
Comment (chaaaaun) [reply src/test/java/scrolls/elder/model/log/LogTest.java:60]: Good point, I've added the test

**Review by CJerrong — APPROVED** [review]
LGTM, I may have to refactor a few of the model test cases after adding the undo, redo commands, should be all good.

**Pull request merged**